### ISSUE TITLE
Setting toolbarVisible is ignored when initial animations are not finished on iOS

### ISSUE THREAD

**Issue by patrykmol** [eclipsesource/tabris-js#597]
Initially setting the property to `false` doesn't have any effect. It must be set in the resize event of the page for it to work.

Workaround:
```
tabris.create("Page", {
  title: "Creating a fullscreen page",
  topLevel: true
}).on("resize", function() {
  tabris.ui.set("toolbarVisible", true);
}).open();

tabris.ui.set("toolbarVisible", false);
```

**Comment by cpetrov** [comment]
This workaround does not work for non-topLevel pages. The top bar gets shortly black and then reappears again.

```javascript
var page = tabris.create("Page", {
  title: "Creating a new page dynamically",
  topLevel: true
});

tabris.create("Button", {
  layoutData: {left: 10, top: 10, right: 10},
  text: "Create and open a new page"
}).on("select", function() {
  tabris.create("Page", {
    title: "Dynamically created page",
    topLevel: false
  }).once("resize", function() {
    tabris.ui.set("toolbarVisible", false);
  }).once("disappear", function() {
    tabris.ui.set("toolbarVisible", true);
  }).open();
}).appendTo(page);

page.open();
```

**Comment by bFlood** [comment]
any update on this? how can you get the toolbar to not display for non-topLevel pages?

**Comment by patrykmol** [comment]
No progress at the moment. You could try delaying the call with timer.

**Comment by bFlood** [comment]
even if this works, when you go back to the main page, the toolbar is visible. this seems pretty broken at the moment

will this ever work? will we be able to at least turn off the animation?

**Comment by patrykmol** [comment]
This really falls to Apple. The problem here is that the animation is done by the system and there's nothing we could do about it.

**Comment by bFlood** [comment]
thanks @patrykmol 

doesn't this method allow you to turn off animation?
https://developer.apple.com/library/ios/documentation/UIKit/Reference/UINavigationController_Class/#//apple_ref/occ/instm/UINavigationController/setNavigationBarHidden:animated:

**Comment by patrykmol** [comment]
No. It only tells system if you want to hide/show navbar with animation or not. The animation that is the problem here is triggered by iOS itself and there isn't much we can do to work around it.

**Comment by bFlood** [comment]
how do other iOS apps with custom navbar deal with the issue then? (say, Google Maps)

and thanks for the info, I'll try to work it out with timers

**Comment by patrykmol** [comment]
As you said it. They have custom navigation. :)

**Comment by bFlood** [comment]
exactly, how do keep the native navbar completely hidden in a tabris app? for this particulat app, that's exactly what I need however when I open a non-topLevel page, the toolbar appears

**Comment by patrykmol** [comment]
They don't use it at all. You don't have to use navigation controller in an iOS app. However due to the framework design we use navigation controller as root controller for the app.

**Comment by bFlood** [comment]
ok, I understand and that makes sense.

**Comment by bFlood** [comment]
the timer workaround does work however, the visual of the navbar displaying and hiding is when transitioning between pages is really bad. not to mention a quick swipe backwards can leave leave the toolbar area completely black

so, I guess for now this type of app (no navbar) is just not possible in tabris right now. I'd love to see this in future version (as well as a more customizable navbar)

**Comment by bev-on** [comment]
@patrykmol @cpetrov @bFlood as I already mentioned in another [issue](https://github.com/eclipsesource/tabris-js/issues/897), I find solution/workaround for this problem.

```style: ["FULLSCREEN"]``` as page property, will fix this issue.
```javascript
tabris.ui.set("toolbarVisible", false);

var page = new tabris.Page({
    title: "Custom Actionbar",
    style: ["FULLSCREEN"],
    topLevel: true
});
```
Tested and working on both iOS and Android without any timers and onresize listeners... and isn't affected by page changes or page close. If you set ```style: ["FULLSCREEN"]``` to page you want to open, there won't be visible toolbar/actionbar.
